lib/components/viewers/stop-viewer.js

```diff
@@ -423,19 +424,23 @@ function getFormattedStopTime (stopTime, homeTimezone, soonText = 'Due', timeFor
 
   const now = moment().tz(homeTimezone)
   const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
-  const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone)
-  const differentDay = now.date() !== serviceDay.date()
-
+  const arrivesAfterMidnight = stopTime.realtimeDeparture >= ONE_DAY_IN_SECONDS
+  // Calculate current home seconds after midnight. If arrival occurs after
+  // midnight, adjust the value by one day in seconds.
+  const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone) + (arrivesAfterMidnight ? ONE_DAY_IN_SECONDS : 0)
```

Comment:
This won't work for stop times that occur more than 1 day after the service date. To resolve this you could do something like this:

```javascript
const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone) + (stopTime.realtimeDeparture - (stopTime.realtimeDeparture % ONE_DAY_IN_SECONDS))
```

Reply:
Again, I'm not understanding when a stop time would ever be more than 1 day after the service date https://github.com/opentripplanner/otp-react-redux/pull/98#discussion_r313905657

Reply:
Actually, I think this is incorrect (both mine and yours). I've got a fix coming in though based on your other comment.